crates/teloxide-core/src/types/message_reaction_updated.rs

```diff
@@ -15,12 +15,11 @@ pub struct MessageReactionUpdated {
     #[serde(flatten)]
     pub message_id: MessageId,
 
-    /// The user that changed the reaction, if the user isn't anonymous
-    pub user: Option<User>,
-
-    /// The chat on behalf of which the reaction was changed, if the user is
-    /// anonymous
-    pub actor_chat: Option<Chat>,
+    /// The [`MaybeAnonymousUser::User`] that changed the reaction, if the user
+    /// isn't anonymous or the [`MaybeAnonymousUser::Chat`] on behalf of
+    /// which the reaction was changed, if the user is anonymous
+    #[serde(deserialize_with = "deserialize_actor", flatten)]
+    pub actor: MaybeAnonymousUser,
```

Comment:
What's your opinion on changing the field's name compared to the official TBA version with separate `user` and `actor_chat`?

TBA doesn't have the `actor` field, so this change essentially trades consistency with TBA for type safety and "Rust-way" design.

Do we have other examples in the API that do the same thing?

Reply:
Another one is changed in this PR too (`PollAnswer::voter`). They are really the case for some enumeration because either of them is present at any time.

I'm not sure with a proper name for that field (it could be named plainly `user` as well).

Similar thing TBA does with `MaybeInaccessibleMessage`, so here we are somewhat consistent with this style

Reply:
And we did this with the `InputSticker` previously (now changed, , which was just an enumeration over the optional types for file formats

Reply:
We do this all the time. Actually, there are not many cases for `two separate mutually exclusive` fields except those I mentioned above, but it is what we actually do when we have `mutually exclusive` fields (MessageCommon, MediaKind, etc..).

Personally, I don't see any problem with providing exclusive api and field renaming (everything is covered by docs), when only one variant is supposed, even when there are only two variants

Reply:
And semantically TBA docs always focused only on `JSON`, so we can do nothing with it except producing new fields and rename others to make better semantic with strong type system

Reply:
I like "Rust-way" more too. Maybe just mention in doc comment about mismatch with TBA JSON?

Reply:
So we did this kind of renaming previously: we renamed [`PollAnswer`](https://core.telegram.org/bots/api#pollanswer)'s `voter_chat` and `user` to a single [`PollAnswer::voter`](https://docs.rs/teloxide/latest/teloxide/prelude/struct.PollAnswer.html#structfield.voter).

So it seems that this change aligns well with our previous decisions.